### PR TITLE
feat: onboard IST with elements button

### PR DESCRIPTION
**Pull request by samsiegart**
refs https://github.com/Agoric/ui-kit/issues/78

This PR bases off https://github.com/Agoric/ui-kit/pull/97, will move out of draft after that merges.

This will eventually be used in the provision-wallet-notice dialog (future PR), but can also be used standalone.

It uses the `interchain-ui` button component so it's themeable with the `ThemeProvider` and looks consistent with the network dropdown and wallet connection modal.

Also exposes `leapElementsClient` if clients need to use leap elements in a different context.

## Testing

Tested with dapp-offer-up in https://github.com/Agoric/dapp-offer-up/commit/55946aed0a3ac1250b4d248a38956c4c7fecced1

## Screenshots

image

image

### CODE REVIEW

File: packages/react-components/src/lib/config/index.ts [hunk 1]
```diff
@@ -1,7 +1,9 @@
 import { assets, chains } from 'chain-registry';
-import type { Chain, AssetList } from '@chain-registry/types';
 import type { Endpoints } from '@cosmos-kit/core';
 
+type Chain = (typeof chains)[number];
+type AssetList = (typeof assets)[number];
```
Comment (turadg) [inline packages/react-components/src/lib/config/index.ts:5]: import { AssetList, Chain } from '@chain-registry/types';

File: packages/web-components/src/wallet-connection/watchWallet.ts [hunk 2]
```diff
@@ -33,6 +33,8 @@ type VBankAssets = [
 
 /** @typedef {import('@agoric/ertp/src/types.js').Amount<'nat'>['value']} NatValue */
 
+/** @typedef {import('@agoric/ertp/src/types.js').Amount<'nat'>['value']} NatValue */
```
Comment (turadg) [inline packages/web-components/src/wallet-connection/watchWallet.ts:36]: this is a `.ts` file so the jsdoc tyepdefs do nothing. 

this and the one above should be removed. Line 19 actually imports the type

File: packages/react-components/src/lib/config/index.ts (hunk 1 above)
Comment (samsiegart) [reply packages/react-components/src/lib/config/index.ts:5]: I was seeing some type mismatch and resorted to this, but tried it again and it seems to build 👍

File: packages/react-components/src/lib/utils/leapElementsClient.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import type { WalletClient } from '@leapwallet/elements';
+import { useWalletClient } from '@cosmos-kit/react';
+import { useMemo } from 'react';
+
+export const useElementsWalletClient = (): WalletClient => {
+  const { client } = useWalletClient();
+
+  // @ts-expect-error Mismatch between `Long` type in `signDoc`
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply packages/react-components/src/lib/utils/leapElementsClient.ts:8]: This is without the expect-errors, I needed to put them in both places:

```
$ yarn build
src/lib/utils/leapElementsClient.ts:8:9 - error TS2322: Type '{ enable: (chainIds: string | string[]) => Promise<void>; getAccount: (chainId: string) => Promise<{ bech32Address: string; pubKey: Uint8Array; isNanoLedger: boolean; }>; getSigner: (chainId: string) => Promise<...>; }' is not assignable to type 'WalletClient'.
  The types returned by 'getSigner(...)' are incompatible between these types.
    Type 'Promise<{ signDirect: (address: string, signDoc: SignDoc) => Promise<{ signature: Uint8Array; signed: SignDoc; }>; signAmino: (address: string, signDoc: StdSignDoc) => Promise<...>; }>' is not assignable to type 'Promise<Signer>'.
      Type '{ signDirect: (address: string, signDoc: SignDoc) => Promise<{ signature: Uint8Array; signed: SignDoc; }>; signAmino: (address: string, signDoc: StdSignDoc) => Promise<...>; }' is not assignable to type 'Signer'.
        The types returned by 'signDirect(...)' are incompatible between these types.
          Type 'Promise<{ signature: Uint8Array; signed: SignDoc; }>' is not assignable to type 'Promise<SignDirectResponse | TxRaw>'.
            Type '{ signature: Uint8Array; signed: SignDoc; }' is not assignable to type 'SignDirectResponse | TxRaw'.
              Type '{ signature: Uint8Array; signed: SignDoc; }' is not assignable to type 'SignDirectResponse'.
                The types of 'signed.accountNumber' are incompatible between these types.
                  Type 'bigint' is not assignable to type 'Long'.

8   const walletClient: WalletClient = useMemo(() => {
          ~~~~~~~~~~~~

src/lib/utils/leapElementsClient.ts:28:61 - error TS2345: Argument of type 'import("/home/samsiegart/ui-kit/node_modules/@leapwallet/elements-core/node_modules/cosmjs-types/cosmos/tx/v1beta1/tx").SignDoc' is not assignable to parameter of type 'import("/home/samsiegart/ui-kit/node_modules/cosmjs-types/cosmos/tx/v1beta1/tx").SignDoc'.
  Types of property 'accountNumber' are incompatible.
    Type 'Long' is not assignable to type 'bigint'.

28             const result = await signer.signDirect(address, signDoc);
                                                               ~~~~~~~


Found 2 errors in the same file, starting at: src/lib/utils/leapElementsClient.ts:8
```

So seems like the `cosmjs-types` in leap and cosmos-kit differ. I'm not sure how to cast in this scenario, but the errors seem inconsequential.

File: packages/web-components/src/wallet-connection/watchWallet.ts (hunk 2 above)
Comment (samsiegart) [reply packages/web-components/src/wallet-connection/watchWallet.ts:36]: Oh yea this seems to be a merge artifact, removed.

File: packages/react-components/src/lib/icons/Ist.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+export const Ist =
```
[the remark this answers is not in the document]
Comment (samsiegart) [reply packages/react-components/src/lib/icons/Ist.ts:2]: You can copy the data uri of an image in chrome dev tools or with various web apps. Just added a comment.

**Review by turadg — APPROVED** [review]

**Pull request merged**